classes/Configuration.php

```diff
@@ -438,6 +438,14 @@ public static function updateGlobalValue($key, $values, $html = false)
      */
     public static function updateValue($key, $values, $html = false, $idShopGroup = null, $idShop = null)
     {
+        if (!is_array($values)) {
+            $values = [$values];
+        }
+        foreach ($values as $lang => $value) {
+            $storedValue = Configuration::get($key, $lang, $idShopGroup, $idShop, null);
+            Hook::exec('actionConfigurationUpdateBefore', ['name' => $key, 'current_value' => $storedValue, 'new_value' => $value]);
+        }
```

Comment:
I think this could be a better way, passing all parameters. If the hook subscriber needs the info, he can get it manually. Also, don't forget to include this hook in the install fixtures and autoupgrade.

```suggestion
        Hook::exec('actionConfigurationUpdateValueBefore', [
          'key' => $key,
          'values' => $values,
          'html' => $html,
          'idShopGroup' => $idShopGroup,
          'idShop' => $idShop,
        ]);
```